src/openforms/formio/api/validators.py

```diff
@@ -37,8 +37,12 @@ def __call__(self, value: UploadedFile) -> None:
                 ),
             )
 
-        # Contents is allowed. Do extension or submitted content_type agree?
+        # gh #2520
+        if mime_type == "application/CDFV2":
+            whole_file = head + value.read()
```

Comment:
The file size validation is done before the mime type validation, so if it's too large a file it should have already failed validation